src/gurt/misc.c

```diff
@@ -11,12 +11,41 @@
 
 #include <stdarg.h>
 #include <math.h>
+#include <malloc.h>
 #include <string.h>
 #include <gurt/common.h>
 
 /* state buffer for DAOS rand and srand calls, NOT thread safe */
 static struct drand48_data randBuffer = {0};
 
+d_alloc_track_cb_t d_alloc_track_cb;
+d_alloc_track_cb_t d_free_track_cb;
+static __thread void *track_arg;
+
+#define D_MEMORY_TRACK_ENV "D_MEMORY_TRACK"
+
+void
+d_set_alloc_track_cb(d_alloc_track_cb_t alloc_cb, d_alloc_track_cb_t free_cb)
+{
+	bool	track_mem = false;
+
+	d_getenv_bool(D_MEMORY_TRACK_ENV, &track_mem);
```

Comment:
You can't read an engine specific env var in common gurt code. Please move this check to engine and also make sure all the related metrics are only created when this "D_MEMORY_TRAC" is enabled.

Reply:
sure